ktor-io/jvm/src/io/ktor/utils/io/core/ByteReadPacketExtensions.jvm.kt

```diff
@@ -18,7 +18,9 @@ public fun Source.readAvailable(buffer: ByteBuffer): Int {
 }
 
 public fun Source.readFully(buffer: ByteBuffer) {
-    readAtMostTo(buffer)
+    while (!exhausted()) {
+        readAtMostTo(buffer)
```

Comment:
Please check the case when `buffer.remaining() == 0`, having a unit test would be great

Reply:
@e5l thanks, fixed.